src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +33,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream
+                .range(0, numbers.size())
+                .mapToObj(i -> (i % 2) != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(x -> x % 2 != 0)
+                .mapToDouble(x -> x).average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
```suggestion
                .orElseThrow();
```

what `orElseThrow()` method throws by default?